skorch/tests/test_helper.py

```diff
@@ -708,3 +711,183 @@ def test_describe_signature_other_dtypes(self, transformer_cls, df):
             'col_cats': {"dtype": torch.int32, "input_units": 2},
         }
         assert result == expected
+
+
+class TestAccelerate:
+    @pytest.fixture(scope='module')
+    def data(self, classifier_data):
+        return classifier_data
+
+    @pytest.fixture(scope='module')
+    def module_cls(self, classifier_module):
+        return classifier_module
+
+    @pytest.fixture
+    def accelerator_cls(self):
+        pytest.importorskip('accelerate')
+
+        from accelerate import Accelerator
+
+        return Accelerator
+
+    @pytest.fixture
+    def net_cls(self, module_cls):
+        from skorch import NeuralNetClassifier
+        from skorch.helper import AccelerateMixin
+
+        class AcceleratedNet(AccelerateMixin, NeuralNetClassifier):
+            pass
+
+        return partial(
+            AcceleratedNet,
+            module=module_cls,
+            max_epochs=2,
+            lr=0.1,
+            device=None,
+        )
+
+    @pytest.mark.parametrize('mixed_precision', ['no', 'fp16', 'bf16'])
+    def test_mixed_precision(self, net_cls, accelerator_cls, data, mixed_precision):
+        # Only test if training works at all, no specific test of whether the
+        # indicated precision is actually used, since that depends on the
+        # underlying hardware.
+        import accelerate
+
+        if LooseVersion(accelerate.__version__) > '0.5.1':
+            accelerator = accelerator_cls(mixed_precision=mixed_precision)
+        elif mixed_precision == 'bf16':
+            pytest.skip('bf16 only supported in accelerate version > 0.5.1')
+        else:
+            fp16 = mixed_precision == 'fp16'
+            accelerator = accelerator_cls(fp16=fp16)
+
+        net = net_cls(accelerator=accelerator)
+        X, y = data
+        net.fit(X, y)  # does not raise
```

Comment:
The following assertion fails here:

```suggestion
        net.fit(X, y)  # does not raise
        assert not np.isnan(net.history[:, "train_loss"]).all()
```

(I have not debugged this yet)

Reply:
ping

Reply:
Turning on `fp16` results in weights being nan. I think we can debug this on a CPU:

```python
    def test_force_cpu(self, net_cls, accelerator_cls, data):
        accelerator = accelerator_cls(cpu=True)
        net = net_cls(accelerator=accelerator)
        net.fit(*data)  # does not raise
        
        # this fails
        assert np.isfinite(net.history[:, "train_loss"]).all()
```

Reply:
Looks like everything works now!